astropy/table/table.py

```diff
@@ -1312,9 +1312,16 @@ def __setitem__(self, item, value):
     def __delitem__(self, item):
         if isinstance(item, six.string_types):
             self.remove_column(item)
-        elif isinstance(item, tuple):
+
+        elif isinstance(item, (six.integer_types, np.integer)):
```

Comment:
You can use ``_is_int`` from ``astropy.io.fits.utils`` to check for integer types